.github/workflows/release.yml

```diff
@@ -0,0 +1,122 @@
+name: Docker and Helm Release
+
+on: workflow_dispatch
+
+permissions:
+  contents: write
+  issues: read
+  pull-requests: read
+  packages: write
+
+jobs:
+  release:
+    name: Validate Docs, Tag, and Docker Push & Helm Push
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.BOT_TOKEN }}
+
+      - name: Get New Version
+        id: nv
+        run: |
+          apt-get update && apt-get install yq
+          version=$(yq .version helm-charts/vmpooler/Chart.yaml)
+          appVersion=$(yq .appVersion helm-charts/vmpooler/Chart.yaml)
+          echo "version=$version" >> $GITHUB_OUTPUT
+          echo "appVersion=$appVersion" >> $GITHUB_OUTPUT
+          echo "Found version $version from helm-charts/vmpooler/Chart.yaml"
+          echo "Found appVersion $version from helm-charts/vmpooler/Chart.yaml"
+
+      - name: Get Current Version
+        uses: actions/github-script@v6
+        id: cv
+        with:
+          script: |
+            const { data: response } = await github.rest.repos.getLatestRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            })
+            console.log(`The latest release is ${response.tag_name}`)
+            return response.tag_name
+          result-encoding: string
+
+      - name: Generate Changelog
+        uses: docker://githubchangeloggenerator/github-changelog-generator:1.16.2
+        with:
+          args: >-
+            --future-release ${{ steps.nv.outputs.version }}
+        env:
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Validate Changelog
+        run : |
+          set -e
+          if output=$(git status --porcelain) && [ ! -z "$output" ]; then
```

Comment:
Trivial: this can be written more concisely:

    if [[ -n $(git status --porcelain) ]]; then

Reply:
Hm, I don't know how I originally ended up with that, but yes thanks that is much better.